### PR TITLE
Refactor core and components

### PR DESCRIPTION
**Pull request by lferraz**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: limbus/core/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from .core import ComponentsManager, Component, DefaultParam, ComponentState, Params, NoValue
```
Comment (edgarriba) [inline limbus/core/__init__.py:1]: No big deal but what about having component.py and manager.py ?

File: limbus/core/factory.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+            _add_modules_to_globals(comp_globals, ["collections"])
+            named_tpl = (f"collections.namedtuple('{tp}', {returns}.keys(),"
+                         f"defaults=list(map(eval, {returns}.values())))")
+        # the new type must be in the globals to be used
+        comp_globals[tp] = eval(named_tpl, comp_globals)
+        return tp
+
+
+def register_components(comp_globals: Dict[str, Any], lst_components: List[ComponentDefinition]) -> None:
+    """Register all teh components of a list of components.
```
Comment (edgarriba) [inline limbus/core/factory.py:128]:
```suggestion
    """Register all the components of a list of components.
```

File: limbus/core/__init__.py (hunk 1 above)
Comment (lferraz) [reply limbus/core/__init__.py]: done, please, check it

File: limbus/core/core.py [hunk 3]
```diff
@@ -1,24 +1,12 @@
-"""Core methods to manage components."""
```
Comment (edgarriba) [inline limbus/core/core.py]: This is why I suggested to  name this module component.py

File: limbus/core/manager.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+            assert isinstance(to_link, _Link)
+            to_link = {from_idx: to_link}
+        if to_idx is not None:
+            assert isinstance(from_link, _Link)
+            from_link = {to_idx: from_link}
+        self.nodes[_from.name].outputs[from_name].append(to_link)
+        self.nodes[_to.name].inputs[to_name].append(from_link)
+
+    def _traverse(self, node_name) -> None:
+        # remainder about teh pins:
```
Comment (edgarriba) [inline limbus/core/manager.py:106]:
```suggestion
        # remainder about the pins:
```

File: limbus/core/__init__.py [hunk 5]
```diff
@@ -1,4 +1,4 @@
-from limbus.core.core import Component, ComponentState, Params, NoValue
+from limbus.core.component import Component, ComponentState, Params, NoValue
```
Comment (edgarriba) [inline limbus/core/__init__.py:1]: :smile:

Comment (lferraz) [reply limbus/core/__init__.py:1]: 😝

**Pull request merged**